plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/config.py

```diff
@@ -7,65 +7,18 @@
 from hydra.types import ObjectConf
 
 
-class QueueType(Enum):
-    auto = "auto"
+class ExecutorName(Enum):
     local = "local"
     slurm = "slurm"
 
 
 @dataclass
-class SlurmQueueConf:
-    # Params are used to configure sbatch, for more info check:
-    # https://github.com/facebookincubator/submitit/blob/master/submitit/slurm/slurm.py
-
-    # maximum time for the job in minutes
-    time: int = 60
-    # number of cpus to use for each task
-    cpus_per_task: int = 10
-    # number of gpus to use on each node
-    gpus_per_node: int = 1
-    # number of tasks to spawn on each node
-    ntasks_per_node: int = 1
-    # number of nodes to use for the job
-    nodes: int = 1
-    # memory to reserve for the job on each node, in GB
-    mem: str = "${hydra.launcher.mem_limit}GB"
-    # slurm partition to use on the cluster
-    partition: Optional[str] = None
-    # USR1 signal delay before timeout
-    signal_delay_s: int = 120
-    # name of the job
-    job_name: str = "${hydra.job.name}"
-    # Maximum number of retries on job timeout.
-    # Change this only after you confirmed your code can handle re-submission
-    # by properly resuming from the latest stored checkpoint.
-    # check the following for more info on slurm_max_num_timeout
-    # https://github.com/facebookincubator/submitit/blob/master/docs/checkpointing.md
-    max_num_timeout: int = 0
-
-
-@dataclass
-class LocalQueueConf:
-    # local executor mocks the behavior of slurm locally
-
-    # maximum time for the job in minutes
-    timeout_min: int = 60
-    # number of gpus to use on each node
-    gpus_per_node: int = 1
-    # number of tasks to spawn on each node (only one node available in local executor)
-    tasks_per_node: int = 1
+class BaseParams:
+    """Configuration shared by all executors
+    """
 
-
-@dataclass
-class AutoQueueConf:
-    # auto executor automatically identifies and uses available cluster
-    # Currently this is only slurm, but local executor can be manually forced
-    # instead.
-    # Most parameters are shared between clusters, some can be cluster specific
-
-    # cluster to use (currently either "slurm" or "local" are supported,
-    # None defaults to an available cluster)
-    cluster: str = "slurm"
+    executor: ExecutorName = ExecutorName.slurm  # can we get rid of this?
+    submitit_folder: str = "${hydra.sweep.dir}/.${hydra.launcher.params.executor.value}"
 
     # maximum time for the job in minutes
     timeout_min: int = 60
```

Comment:
we'll need to decide if we update some of the names for more clarity (timeout_min-> timeout_minute(s)?) or forward compatibility (gpus_per_node-> gpus_per_task?)

Reply:
for me timeout_min is just as clear as timeout_minutes.

As for future compatibility: your call.
you will be able to transition from gpus_per_node-> gpus_per_task by manually deprecating it:

1. add support for both and a comment that gpus_per_node is deprecated in the config.
2. if the user provide gpus_per_node, issue a UserWarning and convert it to gpus_per_task, if the user provides gpus_per_task  use that.
3. after a few months we can remove gpus_per_node completely.

This strategy may be too painful to implement for a plugin, but this is what I am trying to do with Hydra itself.
The alternative is to just have a release that is breaking it at once, which will have users coming with pitchforks asking about how to fix it.

Reply:
> for me timeout_min is just as clear as timeout_minutes.

Then lets keep it that way

> you will be able to transition from gpus_per_node-> gpus_per_task by manually deprecating it:

If that kind of transition is fine for you, since it's basically what will happen in the submitit package it will be definitely easy to handle here. So let's keep it to what's in the package then.

Reply:
absolutely.
we can support both temporarily and issue a warning if someone uses the old one until we are comfortable nuking it.
Now is a chance to do some serious clean up but normally we will want to maintain compatibility.